### PR TITLE
MP-138 Collection card and modal updates

### PR DESCRIPTION
**Pull request by LeoMcA**
- update collections card styling
- clamp displayed lines
- update collection create/edit modal
- change description input to textarea
- add auto-expanding textarea component
- add character limited input/textarea component

screenshot

screenshot

### CODE REVIEW

**Comment by LeoMcA** [comment]
Some small changes: rebased against main, removed line-clamp (I'll do this in MP-143 instead, to ensure we don't clamp collection items, so this commit can be released independent of other design changes), ensured textarea whitespace is preserved when displaying descriptions (and removed some unnecessary css while touching those selectors)

File: client/src/plus/collections/new-edit-collection-modal.tsx [hunk 1]
```diff
@@ -44,7 +46,9 @@ export default function NewEditCollectionModal({
   const { isPending, resetError, error } =
     "id" in collection ? editHook : createHook;
 
-  const changeHandler = (e: React.ChangeEvent<HTMLInputElement>) => {
+  const changeHandler: React.ChangeEventHandler<
+    HTMLInputElement | HTMLTextAreaElement
+  > = (e) => {
     const { name, value } = e.target;
     setCollection({ ...collection, [name]: value.trimStart() });
```
Comment (Guyzeroth) [inline client/src/plus/collections/new-edit-collection-modal.tsx:53]: trimStart seems to make editing from the start of the text area impossible.

**Review by Guyzeroth — COMMENTED** [review]
LGTM one non-regression issue I just noticed with trimStart

**Review by Guyzeroth — APPROVED** [review]
Looks nice. I like the UX better now.

**Pull request merged**